docs(readme): autumn cleaning (#1341)
ipfs-shipyard/ipfs-companion: README.md

## About IPFS Companion

IPFS Companion harnesses the power of your locally running IPFS [Kubo](https://github.com/ipfs/kubo) node (either through the [IPFS Desktop](https://docs.ipfs.tech/install/ipfs-desktop/) app or the [command-line daemon](https://docs.ipfs.tech/install/command-line/)) directly inside your favorite Chromium-based or Firefox browser, enabling support for [`ipfs://` addresses](https://docs.ipfs.tech/how-to/address-ipfs-on-web/#native-urls), redirecting content-addressed websites and file paths to your local [Gateway](https://docs.ipfs.tech/concepts/glossary/#gateway), easy IPFS file import and sharing, and more.

> IPFS is a peer-to-peer hypermedia protocol designed to make the web faster, safer, more resilient, and more open. It enables completely distributed sites and applications that don’t rely on centralized hosting and stay true to the original vision of an open, flat web. Visit [the IPFS Project website](https://ipfs.tech) to learn more.

Reviewer: maybe a callout to ipshipyard.com could be placed here as well?

or "maintained by ipshipyard" somewhere?

Author: good idea, added in 7630978969c41615f71500506b1199c7afe533e6

## IPFS Companion features

### Automatically detect and redirect IPFS resources

#### Detect URLs with IPFS paths

> IPFS Companion detects and tests requests for IPFS-like paths (such as `/ipfs/{cid}` or `/ipns/{peerid_or_host-with-dnslink}`) on any website. If a path is a [valid IPFS address](https://github.com/ipfs/is-ipfs), it is redirected to load from your local gateway. The gateway at `localhost` will also automatically switch to a [subdomain gateway](https://docs.ipfs.tech/how-to/address-ipfs-on-web/#subdomain-gateway) to provide a unique origin for each website:

Reviewer: a little more direct?

```suggestion
IPFS Companion detects requests for IPFS-like paths (such as `/ipfs/{cid}` or `/ipns/{peerid_or_host-with-dnslink}`) on any website. If a path is a [valid IPFS address](https://github.com/ipfs/is-ipfs), it is redirected to load from your local gateway. The gateway at `localhost` will also automatically switch to a [subdomain gateway](https://docs.ipfs.tech/how-to/address-ipfs-on-web/#subdomain-gateway) to provide a unique origin for each website:
```

> [!NOTE]
> **Path gateway redirect flow:**
> 
> - **Step 1:** Public gateway URL detected
>   - `https://ipfs.io/ipfs/QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR`
> - **Step 2:** Redirect to local gateway
>   - `http://localhost:8080/ipfs/QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR`
> - **Step 3:** Automatic upgrade to subdomain with origin isolation
>   - `http://bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi.ipfs.localhost:8080`

> [!NOTE]
> **Subdomain gateway redirect flow:**
> 
> - **Step 1:** Subdomain gateway URL detected
>   - `https://bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi.ipfs.dweb.link`
> - **Step 2:** Redirect to local subdomain with origin isolation
>   - `http://bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi.ipfs.localhost:8080`

#### Detect DNSLink-enabled URLs

> IPFS Companion [detects DNSLink info](https://docs.ipfs.tech/how-to/dnslink-companion/) in the DNS records of websites. If a site uses DNSLink (a few examples are https://docs.ipfs.tech, https://ipld.io, and http://tr.wikipedia-on-ipfs.org), IPFS Companion redirects the HTTP request to your local gateway:

Reviewer: and vitalik.eth? it could be good to call out some other major players instead of only links we own.

Author: Let's keep as-is for this PR:
- `.eth` introduces world of pain and UX papercuts:
  - non-ICANN, `http://vitalik.eth` does not work  as expected  (no redirect to local gateway) because 
    - ICANN versions  like `_dnslink.vitalik.eth.limo` and `_dnslink.vitalik.eth.link` do not return any TXT records, which means companion cant detect they are on IPFS and cant redirect to local gateway without extra ENS-specific and domain-specific code

> [!NOTE]
> **DNSLink redirect flow:**
> 
> - **Step 1:** DNSLink-enabled website detected
>   - `http://docs.ipfs.tech`
> - **Step 2:** Redirect to local gateway
>   - `http://localhost:8080/ipns/docs.ipfs.tech`
> - **Step 3:** Automatic upgrade to subdomain with origin isolation

> >   - `http://docs.ipfs.tech.ipns.localhost:8080/`

Reviewer: ```suggestion
>   - `http://docs-ipfs-tech.ipns.localhost:8080/`
```

IPFS Companion enables you to quickly and easily access common actions from your browser bar with just a few clicks:

> - See how many peers you're connected to at a glance at the cube icon in your browser bar

Reviewer: ```suggestion
- The cube icon in your browser bar shows your live peer count for quick reference.
```

## Help & troubleshooting

### Ask a question

> [!TIP]
> The best place to ask about IPFS Companion (or IPFS in general!) is in the [official IPFS Forums](https://discuss.ipfs.tech/), where you can search past discussions for others who may have had the same questions, too.

### Common troubleshooting steps
These frequently encountered troubleshooting situations may be helpful:
- **Import via right-click does not work in Firefox:** See [this workaround](https://github.com/ipfs/ipfs-companion/issues/227).

> - **HTTP-to-HTTPS redirects fail when using Ghostery:** [Ghostery](https://addons.mozilla.org/en-US/firefox/addon/ghostery/) is known to interfere with HTTP-to-HTTPS redirects, which in some setups breaks websites utilizing public gateways [(more details)](https://github.com/ipfs-shipyard/ipfs-companion/issues/466). Until this is fixed upstream, a workaround is to [whitelist](https://user-images.githubusercontent.com/157609/39089525-5834c104-45c9-11e8-9e17-4459a97e5676.png) affected sites.

Reviewer: ```suggestion
- **HTTP-to-HTTPS redirects fail when using Ghostery:** [Ghostery](https://addons.mozilla.org/en-US/firefox/addon/ghostery/) is known to interfere with HTTP-to-HTTPS redirects, which in some setups breaks websites utilizing public gateways [(more details)](https://github.com/ipfs-shipyard/ipfs-companion/issues/466). Until this is fixed upstream, a workaround is to [allowlist](https://user-images.githubusercontent.com/157609/39089525-5834c104-45c9-11e8-9e17-4459a97e5676.png) affected sites.
```
